sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -55,8 +59,31 @@
  * {@link Row} is an immutable tuple-like schema to represent one element in a {@link PCollection}.
  * The fields are described with a {@link Schema}.
  *
- * <p>{@link Schema} contains the names for each field and the coder for the whole record,
- * {see @link Schema#getRowCoder()}.
+ * <p>{@link Schema} contains the names and types for each field.
+ *
+ * <p>There are several ways to build a new Row object. To build a row from scratch using a schema
+ * object, {@link Row#withSchema} can be used. Schema fields can be specified by name, and nested
+ * fields can be specified using the field selection syntax. For example:
+ *
+ * <pre>{@code
+ * Row row = Row.withSchema(schema)
+ *              .withFieldValue("userId", "user1)
+ *              .withFieldValue("location.city", "seattle")
+ *              .withFieldValue("location.state", "wa")
+ *              .build();
+ * }</pre>
+ *
+ * <p>The {@link Row#fromRow} builder can be used to base a row off of another row. The builder can
+ * be used to specify values for specific fields, and all the remaining values will be taken from
+ * the original row. For example, the following produces a row identical to the above row except for
+ * the location.city field.
+ *
+ * <pre>{@code
+ * Row modifiedRow =
+ *     Row.fromRow(row)
+ *        .withFieldValue("location.city", "tacoma")
```

Comment:
Would this be more readable if it was withAddedFieldValue?

Reply:
I don't think so. We're not really adding anything, we're setting something.

Reply:
Nope, I like the proposed `withFieldValue`.